File: uva_online_judge/156.cpp
#include <bits/stdc++.h>

using namespace std;

int main()
{
	string read;
	vector<string> original;
	vector<string> lower_case;
	vector<string> solution;
	map<string, int> ananagrams;
	
	while (cin >> read and read != "#")
	{
		original.push_back(read);
		
		string aux = "";
		for (int i = 0; i < static_cast<int>(read.length()); i++)
			if (tolower(read[i]) >= 'a' and tolower(read[i]) <= 'z')
				aux += tolower(read[i]);
		
		sort(aux.begin(), aux.end());
		
		lower_case.push_back(aux);
		
		ananagrams[aux]++;
	}
	
	for (int i = 0; i < static_cast<int>(lower_case.size()); i++)
		if (ananagrams[lower_case[i]] == 1)
			solution.push_back(original[i]);
	
	sort(solution.begin(), solution.end());
	
	for (int i = 0; i < static_cast<int>(solution.size()); i++)
		cout << solution[i] << endl;
	
	return 0;
}

File: uva_online_judge/156.py
original = []
lowerCase = []
solution = []
ananagram = dict()

inputt = input()
if inputt == "#":
	exit
read = [x for x in inputt.split(" ")]
while read != '#':
	
	for word in read:
		
		original.append(word)
		aux = ""
		for i in range(len(word)):
			if word[i].isalpha():
				aux += word[i].lower()
				
		aux = ''.join(sorted(aux))
		lowerCase.append(aux)
		
		if aux in ananagram:
			ananagram[aux] += 1
		else:
			ananagram[aux] = 1
	
	inputt = input()
	if inputt == "#":
		break
	
	read = [x for x in inputt.split(" ")]
	
for i in range(len(lowerCase)):
	word = lowerCase[i]
	if ananagram[word] == 1:
		solution.append(original[i])
	
solution.sort()
	
for word in solution:
	print(word)

	
